java/core/src/test/org/apache/orc/TestOrcFilterContext.java

```diff
@@ -225,4 +230,12 @@ public void testRepeatingVector() {
     assertTrue(OrcFilterContext.isNull(vectorBranch, 1));
     assertTrue(OrcFilterContext.isNull(vectorBranch, 2));
   }
+  
+  @Test
+  public void testACIDTable(){
```

Comment:
I would also add an actual filter test with data to ensure that the behavior is as expected.

Reply:
Could you please give an existing test as an example?

Reply:
You can take a look at [TestRowFilteringIOSkip](https://github.com/apache/orc/blob/main/java/core/src/test/org/apache/orc/TestRowFilteringIOSkip.java).

I am not sure what the contract wit hive is in this case, but will be good to have the following scenarios:
* File written with ACID schema and read with ACID schema with filter
* File written with ACID schema and read without ACID schema with filter
* File written with no ACID schema and read with ACID schema with filter

Reply:
@zratkai I can help with the this test too if it is taking a lot of your time. Please do let me know.

Reply:
@pavibhai sorry for the delay, I was on a holiday for some days. I created a filter test for ACID table. I am not sure how the last two case should work:

- File written with ACID schema and read without ACID schema with filter
- File written with no ACID schema and read with ACID schema with filter

 I think the schemas should match or not?

Reply:
> I created a filter test for ACID table.
I will take a look at the test today and drop an update.